mcs/restapi/users_login.go

```diff
@@ -0,0 +1,94 @@
+// This file is part of MinIO Console Server
+// Copyright (c) 2020 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package restapi
+
+import (
+	"errors"
+	"log"
+
+	"github.com/minio/m3/mcs/restapi/sessions"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/swag"
+	"github.com/minio/m3/mcs/models"
+	"github.com/minio/m3/mcs/restapi/operations"
+	"github.com/minio/m3/mcs/restapi/operations/user_api"
+	mcCmd "github.com/minio/mc/cmd"
+)
+
+func registerLoginHandlers(api *operations.McsAPI) {
+	// get login strategy
+	api.UserAPILoginDetailHandler = user_api.LoginDetailHandlerFunc(func(params user_api.LoginDetailParams) middleware.Responder {
+		loginDetails, err := getLoginDetailsResponse()
+		if err != nil {
+			return user_api.NewLoginDetailDefault(500).WithPayload(&models.Error{Code: 500, Message: swag.String(err.Error())})
+		}
+		return user_api.NewLoginDetailOK().WithPayload(loginDetails)
+	})
+	// post login
+	api.UserAPILoginHandler = user_api.LoginHandlerFunc(func(params user_api.LoginParams) middleware.Responder {
+		loginResponse, err := getLoginResponse(params.Body)
+		if err != nil {
+			return user_api.NewLoginDefault(500).WithPayload(&models.Error{Code: 500, Message: swag.String(err.Error())})
+		}
+		return user_api.NewLoginCreated().WithPayload(loginResponse)
+	})
+
+}
+
+var ErrInvalidCredentials = errors.New("invalid credentials")
+
+// login performs a check of credentials against MinIO
+func login(accessKey, secretKey *string) (*string, error) {
+	// Probe the credentials
+	cfg, pErr := mcCmd.BuildS3Config(getMinIOServer(), *accessKey, *secretKey, "", "auto")
```

Comment:
mcCmd can be used as a parameter the same way we do it for  all other apis with MinioClient .That way we can add some tests for this.

Reply:
I don't think we can mock imports as they will be linked at compile time, mcCmd is not an object per se